libraries/AP_Param/AP_Param.cpp

```diff
@@ -1577,6 +1575,44 @@ void AP_Param::reload_defaults_file(bool last_pass)
 #endif
 }
 
+#if AP_FILESYSTEM_FILE_READING_ENABLED
+void AP_Param::load_defaults_file_from_filesystem(const char *default_file, bool last_pass)
+{
+    if (load_defaults_file(default_file, last_pass)) {
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+        printf("Loaded defaults from %s\n", default_file);
+#endif
+    } else {
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+        AP_HAL::panic("Failed to load defaults from %s\n", default_file);
+#else
+        printf("Failed to load defaults from %s\n", default_file);
+#endif
+    }
+}
+#endif  // AP_FILESYSTEM_FILE_READING_ENABLED
+
+#if defined(HAL_HAVE_AP_ROMFS_EMBEDDED_H)
+void AP_Param::load_defaults_file_from_romfs(const char *default_file, bool last_pass)
+{
+    const char *prefix = "@ROMFS/";
+    if (strncmp(default_file, prefix, strlen(prefix)) != 0) {
+        // does not start with ROMFS, do not attempt to retrieve from it
+        return;
+    }
+
+    // filename without the prefix:
+    const char *trimmed_filename = &default_file[strlen(prefix)];
+
+    uint32_t string_length;
+    const uint8_t *text = AP_ROMFS::find_decompress(trimmed_filename, string_length);
+    if (text == nullptr) {
+        return;
+    }
+
+    load_param_defaults((const char*)text, string_length, last_pass);
```

Comment:
missing AP_ROMFS::free() in case it is decompressed to allocated memory

Reply:
Fixed